cvat/apps/dataset_manager/bindings.py

```diff
@@ -812,11 +812,7 @@ def _init_frame_info(self):
                 self._excluded_frames.update(self.db_data.validation_layout.disabled_frames)
 
         if self._required_frames:
-            abs_range = self.abs_range
-            self._required_frames = set(
-                self.abs_frame_id(frame) for frame in self._required_frames
-                if frame in abs_range
-            )
+            self._required_frames = set(frame for frame in self._required_frames if frame in self.rel_range)
```

Comment:
Are you sure that `self._required_frames` initially contains relative numbers? It seems that it comes from the `included_frames` constructor parameter, supplied here:

https://github.com/cvat-ai/cvat/blob/49ec1d175d9a9ce106158a7a7936e3c6622d3672/cvat/apps/quality_control/quality_reports.py#L2320

And `job.segment.frame_set` seems to return absolute frame numbers.

Reply:
Looks like one more bug .. 

On `develop` branch in this place we are trying to find intersection between set with absolute frame numbers `job.segment.frame_set` and relative frame numbers `active_validation_frames`:
![image](https://github.com/user-attachments/assets/b2a472d9-440c-47fd-8c4b-db79db0eb2ee)

Reply:
And the exception occurs:

```
[2024-10-16 17:34:44,154] ERROR rq.worker: [Job quality-check-task-4375-user-1]: exception raised while executing (_check_task_quality)
Traceback (most recent call last):
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/.env/lib/python3.10/site-packages/rq/worker.py", line 1431, in perform_job
    rv = job.perform()
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/.env/lib/python3.10/site-packages/rq/job.py", line 1280, in perform
    self._result = self._execute()
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/.env/lib/python3.10/site-packages/rq/job.py", line 1317, in _execute
    result = self.func(*self.args, **self.kwargs)
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/.env/lib/python3.10/site-packages/silk/profiling/profiler.py", line 161, in wrapped_target
    result = target(*args, **kwargs)
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/quality_control/quality_reports.py", line 2259, in _check_task_quality
    return cls()._compute_reports(task_id=task_id)
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/quality_control/quality_reports.py", line 2316, in _compute_reports
    job_data_providers = {
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/quality_control/quality_reports.py", line 2317, in <dictcomp>
    job.id: JobDataProvider(
  File "/usr/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/quality_control/quality_reports.py", line 566, in __init__
    self.job_data = JobData(
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/dataset_manager/bindings.py", line 755, in __init__
    super().__init__(annotation_ir, self._db_task, **kwargs)
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/dataset_manager/bindings.py", line 292, in __init__
    self._init_frame_info()
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/dataset_manager/bindings.py", line 816, in _init_frame_info
    self._required_frames = set(
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/dataset_manager/bindings.py", line 817, in <genexpr>
    self.abs_frame_id(frame) for frame in self._required_frames
  File "/home/bsekachev/app.cvat.ai/cvat_enterprise/cvat/cvat/apps/dataset_manager/bindings.py", line 313, in abs_frame_id
    raise ValueError("Unknown internal frame id %s" % relative_id)
ValueError: Unknown internal frame id 226
```

Reply:
Additionally, it looks like in case with `gt_pool`, `job.segment.frame_set` returns relative frame numbers
But in this case relative = absolute, because `frame_step` and `start_frame` are lost somewhere :/ 
Maybe it was done intentionally (as does not really make sense during random ordering), probably only @zhiltsov-max knows 

![image](https://github.com/user-attachments/assets/e3c37a8e-0741-4e3b-99cd-dc0f64dc196a)

![image](https://github.com/user-attachments/assets/ef7edb1d-1208-42c5-9ab4-902afa08ca31)

Reply:
In the case of gt pool:

- input start frame and frame step make sense only during task creation, so they are dropped after the task is created (otherwise these values would be invalid)
- the gt job segment has the "range" type, opposed to "specific frames" in simple gt jobs. It means it's a regular section of the whole task frame range [0; size), but it has irregular segment size (includes all the pool frames)
- realtive frames are the same as absolute frames (start frame= 0, step = 1 regardless of the user input)